datastore/src/main/scala/org/allenai/datastore/Datastore.scala

```diff
@@ -38,6 +40,27 @@ import scala.collection.JavaConverters._
   * @param s3   properly authenticated S3 client.
   */
 class Datastore(val name: String, val s3: AmazonS3Client) extends Logging {
+
+  private val random = new Random
+  private def withRetries[T](activity: String, retries: Int = 10)(f: => T): T =
+    if (retries <= 0) {
+      f
+    } else {
+      val sleepTime = random.nextInt(10000) + 10000 // sleep between 10 and 20 seconds
```

Comment:
I don't want to make this more complicated. We've been running with this code as it is for two years now, so it's pretty clear that retries are not an important feature for most people. With something that matters so little, I want to keep this as hidden as I can, so that the features that are important remain as transparent as possible.